packages/dashmate/src/listr/tasks/setup/setupRegularPresetTaskFactory.js

```diff
@@ -48,34 +51,187 @@ function setupRegularPresetTaskFactory(
         },
       },
       {
-        title: 'Set node type',
+        title: 'Node type',
         task: async (ctx, task) => {
           if (ctx.nodeType === undefined) {
             ctx.nodeType = await task.prompt([
               {
                 type: 'select',
+                header: '  Dash network has different node types\n  Blue\n  Green\n  Red\n  We'
+                  + ' should'
+                  + ' explain their purpose and costs\n',
                 message: 'Select node type',
-                choices: NODE_TYPES,
-                initial: NODE_TYPE_MASTERNODE,
+                choices: [
+                  { name: NODE_TYPE_MASTERNODE },
+                  { name: NODE_TYPE_HPMN, message: 'high-performance masternode' },
+                  { name: NODE_TYPE_FULLNODE },
+                ],
+                initial: NODE_TYPE_HPMN,
+              },
+            ]);
+
+            // eslint-disable-next-line no-param-reassign
+            task.output = ctx.nodeType;
+          }
+        },
+        options: {
+          persistentOutput: true,
+        },
+      },
+      {
+        enabled: (ctx) => ctx.nodeType === NODE_TYPE_MASTERNODE || ctx.nodeType === NODE_TYPE_HPMN,
+        task: async (ctx, task) => {
+          ctx.isMasternodeRegistered = await task.prompt([
+            {
+              type: 'toggle',
+              header: 'Tell what it means and what we gonna do in both cases\n',
```

Comment:
No idea if this is accurate :man_shrugging: 
```suggestion
              header: 'If your masternode is already registered, we will import your BLS operator key and use it to generate the transaction to move your masternode to dashmate. If you are registering a new masternode, we will generate a BLS operator key for you.\n',
```